src/lib/caches/Items.js

```diff
@@ -94,22 +100,84 @@ module.exports = class ItemsCache {
       let exact = false;
       filtered = undefined;
       base
-        .filter((item) => item && item[by])
+        .filter((item) => item && this.parseNestedKey(item, bys))
         .forEach((item) => {
-          if (item[by].toLowerCase() === term.toLowerCase()) {
-            filtered = item;
-            exact = true;
+          const values = this.parseNestedKey(item, bys);
+          // parseNestedKey returns an array, so check all values in that array
+          for (const entry of values) {
+            logger.error(entry.toLowerCase());
+            if (entry.toLowerCase() === term.toLowerCase()) {
+              filtered = item;
+              exact = true;
+            }
           }
 
-          if (item[by].toLowerCase().includes(term.toLowerCase()) && !exact) {
-            const distance = item[by].toLowerCase().replace(term.toLowerCase(), '').length;
-            if (!keyDistance || distance < keyDistance) {
-              keyDistance = distance;
-              filtered = item;
+          if (!exact) {
+            // parseNestedKey returns an array, so check all values in that array
+            for (const entry of values.values()) {
+              if (entry.toLowerCase().includes(term.toLowerCase())) {
+                const distance = entry.toLowerCase().replace(term.toLowerCase(), '').length;
+                if (!keyDistance || distance < keyDistance) {
+                  keyDistance = distance;
+                  filtered = item;
+                }
+              }
             }
           }
         });
     }
     return this.#cleanup(filtered, { remove, only });
   }
+
+  /**
+   * Utility function for following nested keys. Since some keys have values that are
+   * an array of non-keyed objects, this function iterates over those arrays and can
+   * return multiple values for a single key path.
+   * @param {Array} item data to search
+   * @param {Array<string>} by array of keys to follow
+   * @returns {Array<string>} or undefined if an invalid path of keys is provided
+   */
+  static parseNestedKey(item, by) {
+    let entry = item;
+    const resultList = [];
+
+    // Walk through keys in order
+    for (let i = 0; i < by.length; i += 1) {
```

Comment:
2) is definitely blasphemy in functional and can cause weird issues

i think 1 is fine, alternatively, i don't know that we _need_ to exit early, since the idea is to get the _closest_ one... so unless you're doing single-result, for which i think i'd fully accept getting all of them and grabbing the first and taking the hit is fine, but i think there's a way to exit early _properly_. i.e. instead of a `forEach` for the top level we can do an `array.every` (which allows you to "break" by returning false)

Reply:
i think do #1 and then I can suggest changes once i have time to pour over it more.... and if #1 is already what you have, i'll test it out locally and see if i can do it with every